Index: libiconv-1.11.1/lib/aliases.h
===================================================================
--- libiconv-1.11.1.orig/lib/aliases.h
+++ libiconv-1.11.1/lib/aliases.h
@@ -398,7 +398,7 @@ static const struct alias aliases[] =
   };
 
 #ifdef __GNUC__
-__inline
+
 #endif
 const struct alias *
 aliases_lookup (register const char *str, register unsigned int len)
Index: libiconv-1.11.1/lib/iconv.c
===================================================================
--- libiconv-1.11.1.orig/lib/iconv.c
+++ libiconv-1.11.1/lib/iconv.c
@@ -161,7 +161,7 @@ static const struct alias sysdep_aliases
 #undef S
 };
 #ifdef __GNUC__
-__inline
+
 #endif
 const struct alias *
 aliases2_lookup (register const char *str)
