From 099575ccf4bae126bcb1c5cdf79b0962f3f66480 Mon Sep 17 00:00:00 2001
From: Ronny Nilsson <ronny.nilsson@iopsys.eu>
Date: Tue, 19 Nov 2019 14:59:30 +0100
Subject: [PATCH 1/2] no coldplug dev remount

---
 plug/coldplug.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/plug/coldplug.c b/plug/coldplug.c
index 12df421..58a4613 100644
--- a/plug/coldplug.c
+++ b/plug/coldplug.c
@@ -43,7 +43,9 @@ void procd_coldplug(void)
 {
 	char *argv[] = { "udevtrigger", NULL };
 	unsigned int oldumask = umask(0);
-
+	/* Why are /dev unmounted and then remounted?!?!
+	   It destroys what has already been populated. */
+#if 0
 	if (!is_container()) {
 		umount2("/dev/pts", MNT_DETACH);
 		umount2("/dev/", MNT_DETACH);
@@ -51,7 +53,7 @@ void procd_coldplug(void)
 		mkdir("/dev/pts", 0755);
 		mount("devpts", "/dev/pts", "devpts", MS_NOEXEC | MS_NOSUID, 0);
 	}
-
+#endif
 	ignore(symlink("/tmp/shm", "/dev/shm"));
 	umask(oldumask);
 	udevtrigger.cb = udevtrigger_complete;
-- 
2.7.4

