From 5fbb6d113e108f566c7fcccee35d9d8a6254b947 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20H=C3=B6glund?= <jonas.hoglund@embeddednation.com>
Date: Tue, 28 Nov 2017 09:57:25 +0100
Subject: [PATCH] Do not include conflicting header file

---
 rtnl_mdb.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/rtnl_mdb.c b/rtnl_mdb.c
index ebe737d..35aa83d 100644
--- a/rtnl_mdb.c
+++ b/rtnl_mdb.c
@@ -35,7 +35,6 @@
 # include "nlattr.h"
 # include "print_fields.h"
 
-# include <netinet/in.h>
 # include <linux/if_bridge.h>
 # include "netlink.h"
 
-- 
2.7.4

